posthog/temporal/batch_exports/squash_person_overrides.py

```diff
@@ -352,18 +370,31 @@ async def select_persons_to_delete(inputs: QueryInputs) -> list[SerializablePers
     from django.conf import settings
 
     latest_created_at = inputs.latest_created_at.timestamp() if inputs.latest_created_at else inputs.latest_created_at
-    to_delete_rows = sync_execute(
-        SELECT_PERSONS_TO_DELETE_QUERY.format(database=settings.CLICKHOUSE_DATABASE),
-        # We pass this as a timestamp ourselves as clickhouse-driver will drop any microseconds from the datetime.
-        # This would cause the latest merge event to be ignored.
-        # See: https://github.com/mymarilyn/clickhouse-driver/issues/306
-        {"latest_created_at": latest_created_at},
-    )
 
-    if not isinstance(to_delete_rows, list):
-        # Could return None if no results or int if this were an insert.
-        # Mostly to appease type checker
-        return []
+    async with get_client() as clickhouse_client:
+        response = await clickhouse_client.read_query(
+            SELECT_PERSONS_TO_DELETE_QUERY.format(database=settings.CLICKHOUSE_DATABASE),
+            # We pass this as a timestamp ourselves as clickhouse-driver will drop any microseconds from the datetime.
+            # This would cause the latest merge event to be ignored.
+            # See: https://github.com/mymarilyn/clickhouse-driver/issues/306
+            query_parameters={"latest_created_at": latest_created_at},
+        )
+
+        if not response:
+            return []
+
+        schema = (
+            int,
+            UUID,
+            UUID,
+            parse_clickhouse_timestamp,
+            int,
+            parse_clickhouse_timestamp,
+        )
+        to_delete_rows = (
+            (schema[field_index](field_value) for field_index, field_value in enumerate(line.split("\t")))
+            for line in response.decode("utf-8").splitlines()
+        )
```

Comment:
Spicy!

Reply:
Yeah, all of this going away in second iteration...